File: array/running-sum-of-1d-array.js
// https://leetcode.com/problems/running-sum-of-1d-array/
// tags - array
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    // const sumsFromLeft = Array(nums.length);
    // for (let i = 0; i < nums.length; i++) {
    //     sumsFromLeft[i] = (i-1 >= 0 ? sumsFromLeft[i-1] : 0) + nums[i];
    // }
    // return sumsFromLeft;
    
    // modify the input array
    for (let i = 1; i < nums.length; i++) {
        nums[i] += nums[i-1];
    }
    return nums;
    
};

File: array/running-sum-of-1d-array.py
# https://leetcode.com/problems/running-sum-of-1d-array/
# tags - array
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        # rsum = []
        # for i in range(len(nums)):
        #     rsum.append(sum(nums[:i+1]))
        # return rsum
        
        # rsum = []
        # sum = 0
        # for num in nums:
        #     rsum.append(num + sum)
        #     sum += num
        # return rsum
    
        # if len(nums) <= 1:
        #     return nums
        # rsum = [nums[0]]
        # for num in nums[1:]:
        #     rsum.append(num + rsum[-1])
        # return rsum
        
        rsum = []
        for i, num in enumerate(nums):
            rsum.append(num + (rsum[-1] if i > 0 else 0))
        return rsum